rcl/src/rcl/wait.c

```diff
@@ -79,6 +80,9 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    if (wait_set->impl->rmw_waitset) {
+      assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
As to whether or not the check should be made, it is a situation where the it should never fail, but because it could in other scenarios there is a warning when you don't check the return value. This check is really defensive and should never be false.